src/api/java/appeng/api/crafting/IPatternDetails.java

```diff
@@ -21,21 +21,39 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-package appeng.api.networking.crafting;
+package appeng.api.crafting;
 
-import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
-public interface ICraftingCallback {
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
 
-    /**
-     * The call back is synchronized with the level you passed as well as all other crafting jobs. This can lead to
-     * deadlocks in cases like submitting new crafting jobs from the callback itself. However it is not limited to this
-     * case and can deadlock in other ones.
-     * 
-     * It will also be called from a separate crafting thread. If your callback is not thread safe you have to take care
-     * if it yourself. For example by using a thread safe queue and handling it with the blockentity/world/server tick.
-     *
-     * @param job - final job
-     */
-    void calculationComplete(@Nonnull ICraftingJob job);
+import appeng.api.storage.data.IAEStack;
+
+public interface IPatternDetails {
+    ItemStack copyDefinition();
+
+    // TODO: is this even useful?
+    boolean isCrafting();
+
+    IInput[] getInputs();
+
+    default IAEStack getPrimaryOutput() {
+        return getOutputs()[0];
+    }
+
+    IAEStack[] getOutputs();
+
+    interface IInput {
+        IAEStack[] getPossibleInputs();
+
+        long getMultiplier();
+
+        boolean isValid(IAEStack input, Level level);
+
+        boolean allowFuzzyMatch();
```

Comment:
We should remove this and roll it directly into the crafting simulation by default.
All it accomplishes is to test for the same item just with different damage values. 
In general damageable items are not very common, so it would be more user friendly to just check if any damaged item fits into the recipe instead of completely enabling every substitute stack.

At least for crafting patterns. Processing ones cannot be verified so we have to stick with the exact item in this case.

Reply:
Removed.